### DISCUSSION TITLE
Upload files via terminal or scan existing files

### DISCUSSION THREAD

**Discussion by mateusznitka** [Forceu/Gokapi#201] [Q&A]
Hi,
Thanks for great soft. 
Is there any option to upload files without web GUI? I mean by some script or something? I would like to just copy files to some location on server and I want this files to appear in gokapi. Like they will be scanned and uploaded somehow.
Or maybe some executable script which scan the directory and upload/register files in gokapi so they could be shared. Script could be saved to cron and problem solved. 
So basically I would like to have the same upload process like in browser but existing in backend.

The whole point of this idea is to share files restored from backups. There are situations that I cannot restore files directly to user machine so I would like use gokapi as temporary share for this files.

Regards, 
MN

**Comment by Forceu** [comment]
Hi, yes there is an API available. In the main menu click on API and then on documentation to see the endpoints. You will need to write a script that scans the folder and then uploads the file through the API
